pkg/pillar/cmd/zedrouter/flowstats.go

```diff
@@ -582,13 +583,19 @@ func DNSMonitor(bn string, bnNum int) {
 	}
 
 	dnssys[bnNum].Done = make(chan bool)
+	dnssys[bnNum].channelOpen = true
 	packetSource := gopacket.NewPacketSource(handle, handle.LinkType())
 	dnsIn := packetSource.Packets()
 	for {
 		var packet gopacket.Packet
 		select {
 		case <-dnssys[bnNum].Done:
 			log.Infof("(FlowStats) DNS Monitor exit on %s(bridge-num %d)", bn, bnNum)
+			dnssys[bnNum].channelOpen = false
+			dnssys[bnNum].Lock()
+			dnsDataRemove(bnNum)
+			dnssys[bnNum].Unlock()
+
 			defer close(dnssys[bnNum].Done)
```

Comment:
The use of defer is unnecessary since you return immediately, and it raises the question of "why"

Reply:
ok. will remove the defer